plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py

```diff
@@ -0,0 +1,312 @@
+            custom_hours = custom_timings.get("hours", DEFAULT_LOOKBACK_HOURS)
+            start = datetime(**custom_date) if custom_date else (dt_now - timedelta(hours=custom_hours))
+            state[LAST_ALERT_TIME] = self.convert_datetime_to_unix(start.strftime(TIME_FORMAT))
+        else:
+            # check if we have held the TS beyond our max lookback
+            lookback_days = custom_config.get(f"{LAST_ALERT_TIME}_days", MAX_LOOKBACK_DAYS)
+            default_date_lookback = dt_now - timedelta(days=lookback_days)  # if not passed from CPS create on the fly
+            custom_lookback = custom_config.get(f"max_{LAST_ALERT_TIME}", {})
+            comparison_date = datetime(**custom_lookback) if custom_lookback else default_date_lookback
+
+            comparison_unix = self.convert_datetime_to_unix(comparison_date.strftime(TIME_FORMAT))
+
+            # Update state if saved time exceeds the lookback limit
+            if comparison_unix > saved_time:
+                saved_time_str = self.convert_timestamp_to_string(saved_time)
+                comparison_str = self.convert_timestamp_to_string(comparison_unix)
+                self.logger.info(f"Saved time {saved_time_str} exceeds cut off, moving to {comparison_str}.")
+                state[LAST_ALERT_TIME] = comparison_unix
+
+        start_time = state.get(LAST_ALERT_TIME)
+        state[QUERY_START_TIME] = state.get(LAST_ALERT_TIME)
+
+        self.logger.info(
+            f"{log_msg}Applying the following start time='{self.convert_timestamp_to_string(start_time)}'."
+        )
+
+        return start_time
+
+    ###########################
+    # Build post body
+    ###########################
+    def build_post_body(self, search_from: int, search_to: int, start_time: int, end_time: int) -> dict:
+        """
+        Helper method to build the post body for the request
+
+        :param search_from: An integer representing the starting offset within the query result set from which you want alerts returned
+        :param search_to: An integer representing the end offset within the result set after which you do not want alerts returned
+        :param filters: An array of filter fields
+
+        :return post_body:
+        """
+        filters = [
+            {"field": self.time_sort_field, "operator": "gte", "value": start_time},
+            {"field": self.time_sort_field, "operator": "lte", "value": end_time},
+        ]
+
+        self.logger.info(
+            f"Query start time: {self.convert_timestamp_to_string(start_time)}, Query end time: {self.convert_timestamp_to_string(end_time)}"
+        )
+
+        post_body = {
+            "request_data": {
+                "search_from": search_from,
+                "search_to": search_to,
+                "sort": {"field": self.time_sort_field, "keyword": "asc"},
+                "filters": filters,
+            }
+        }
+
+        self.logger.info(f"Post Body: {post_body}")
```

Comment:
I don't think we need to log the Post Body as an `info`, better to remove or log as debug

Reply:
Will change this to debug